### PR TITLE
增加自己的模块

### PR DESCRIPTION
**Pull request by yudianer**

### CODE REVIEW

**Comment by lukaszkaiser** [comment]
Can you try again with the CLA? Otherwise I cannot merge it...

File: tensor2tensor/data_generators/translate.py [hunk 1]
```diff
@@ -51,6 +51,153 @@ def vocab_data_files(self):
...
+      while source and target:
+        source_ints = token_vocab.encode(source.strip()) + eos_list
+        target_ints = token_vocab.encode(target.strip()) + eos_list
+        yield {"inputs": source_ints, "targets": target_ints}
+        source, target = source_file.readline(), target_file.readline()
+
+def token_generator_by_source_target(source_path, target_path, source_token_vocab, targe_token_vocab, eos=None):
+  """Generator for sequence-to-sequence tasks that uses tokens.
+
+  自定义（与上面的 token_generator 相对应）：
```
Comment (martinpopel) [inline tensor2tensor/data_generators/translate.py]: English translation?

File: tensor2tensor/data_generators/translate_mnzh.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+FLAGS = tf.flags.FLAGS
+
+# End-of-sentence marker.
+EOS = text_encoder.EOS_ID
+
+
+@registry.register_problem
+class TranslateEnzhWmt32k(translate.TranslateProblem):
+  """
+    Problem spec for Mn_unicode-zh translation.
```
Comment (martinpopel) [inline tensor2tensor/data_generators/translate_mnzh.py]: So is it **en**zh or **mn**zh?

File: tensor2tensor/data_generators/translate_mnzh.py [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+  """
+  @property
+  def targeted_vocab_size(self):
+    return 2**15  # 32k
+
+  @property
+  def source_vocab_name(self):
+    return "vocab.%d.ch.txt" % self.targeted_vocab_size
```
Comment (martinpopel) [inline tensor2tensor/data_generators/translate_mnzh.py]: why "ch"?  ISO_639-1 code for Chinese is "zh".

File: tensor2tensor/data_generators/translate_mnzh.py [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+        target_path: path to the file with target sentences.
+        source_token_vocab: text_encoder.TextEncoder object.
+        target_token_vocab: text_encoder.TextEncoder object.
+        eos: integer to append at the end of each sequence (default: None).
+      Yields:
+        A dictionary {"inputs": source-line, "targets": target-line} where
+        the lines are integer lists converted from tokens in the file lines.
+      """
+    return translate.bi_vocabs_token_generator(data_path + ".mn.shuf",
+                                               data_path + ".ch.shuf",
```
Comment (martinpopel) [inline tensor2tensor/data_generators/translate_mnzh.py]: Again, "ch" is the  ISO 639-1 code for Chamoru language.

File: tensor2tensor/data_generators/translate_mnzh_bpe32k.py [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+from tensor2tensor.utils import registry
+
+import tensorflow as tf
+
+FLAGS = tf.flags.FLAGS
+
+# End-of-sentence marker.
+EOS = text_encoder.EOS_ID
+
+_ENDE_TRAIN_DATASETS = [
```
Comment (martinpopel) [inline tensor2tensor/data_generators/translate_mnzh_bpe32k.py]: Why are ENDE datasets needed in translate_mnzh?

File: tensor2tensor/data_generators/translate_mnzh_bpe32k.py [hunk 6]
```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# Created by Jack on 04/02/2018
+
```
Comment (martinpopel) [inline tensor2tensor/data_generators/translate_mnzh_bpe32k.py]: bpe in name suggests and external BPE dictionary will be downloaded, but this is not the case here

File: .gitignore [hunk 7]
```diff
@@ -1,18 +1,109 @@
-# Compiled python modules.
-*.pyc
+# Byte-compiled / optimized / DLL files
```
Comment (martinpopel) [inline .gitignore]: This is a very user-specific ignore list, more suitable for `git config --global core.excludesfile $HOME/.gitignore` than versioning in this repo.

File: tensor2tensor/data_generators/translate_mnzh.py (hunk 2 above)
Comment (yudianer) [reply tensor2tensor/data_generators/translate_mnzh.py]: It is the translation from Mongolian words to Chinese words.

File: tensor2tensor/data_generators/translate.py (hunk 1 above)
Comment (yudianer) [reply tensor2tensor/data_generators/translate.py]: It is the translation from Mongolian words to Chinese words.

File: tensor2tensor/data_generators/translate_mnzh_bpe32k.py (hunk 5 above)
Comment (yudianer) [reply tensor2tensor/data_generators/translate_mnzh_bpe32k.py]: I just forgot to delete theses code, sorry.

File: .gitignore (hunk 7 above)
Comment (yudianer) [reply .gitignore]: Thank you!

File: tensor2tensor/data_generators/translate_mnzh.py (hunk 4 above)
Comment (yudianer) [reply tensor2tensor/data_generators/translate_mnzh.py]: Sorry, I will make it right.
Thank you!

File: tensor2tensor/data_generators/translate_mnzh.py (hunk 3 above)
Comment (yudianer) [reply tensor2tensor/data_generators/translate_mnzh.py]: Sorry, you are right. I just picked the top two characters of " Chinese " then, and did not think too much about this. Sorry, I will make it right.Thank you.

**Comment by yudianer** [comment]
Thank you for your advice, I think there are too many "
“unsuitable codes". I will try to submit them again after I check them carefully and make them right.

**Comment by lukaszkaiser** [comment]
Still need the CLA from you, could you try again?

**Comment by lukaszkaiser** [comment]
Sorry to not hear back, closing for now.

**Pull request closed without merging**